email-reader/src/test/java/no/sikt/nva/email/reader/handler/ScopusEmailReaderTest.java

```diff
@@ -0,0 +1,182 @@
+        var s3EventNotFromSikt = createS3Event(invalidEmails);
+        var exception = assertThrows(EmailException.class, () -> handler.handleRequest(s3EventNotFromSikt, CONTEXT));
+        assertThat(exception.getMessage(), containsString(COULD_NOT_VERIFY_EMAIL));
+        assertThat(exception.getObjectKey(), is(equalTo(extractObjectKey(s3EventNotFromSikt))));
+        assertThat(exception.getBucket(), is(equalTo(INPUT_BUCKET_NAME)));
+    }
+
+    @Test
+    void shouldThrowExceptionIfEmailDoesNotHaveContentTypeMultipart() throws IOException, MimeException {
+        var s3EventNotFromSikt = createS3Event(EmailGenerator.generateNonMultipartEmail());
+        var exception = assertThrows(EmailException.class, () -> handler.handleRequest(s3EventNotFromSikt, CONTEXT));
+        assertThat(exception.getMessage(), containsString(COULD_NOT_PARSE_EMAIL));
+        assertThat(exception.getObjectKey(), is(equalTo(extractObjectKey(s3EventNotFromSikt))));
+        assertThat(exception.getBucket(), is(equalTo(INPUT_BUCKET_NAME)));
+    }
+
+
+    @Test
+    void shouldThrowExceptionIfEmailDoesNotContainBodyWithURI() throws IOException, MimeException {
+        var s3EventNotFromSikt = createS3Event(EmailGenerator.generateEmailWithoutScopusLinks());
+        var exception = assertThrows(EmailException.class, () -> handler.handleRequest(s3EventNotFromSikt, CONTEXT));
+        assertThat(exception.getMessage(), containsString(NO_URL_PRESENT_IN_MESSAGE));
+        assertThat(exception.getObjectKey(), is(equalTo(extractObjectKey(s3EventNotFromSikt))));
+        assertThat(exception.getBucket(), is(equalTo(INPUT_BUCKET_NAME)));
+    }
+
+    @Test
+    void shouldNotDownloadCitedByUri() throws IOException {
+        var s3EventNotFromSikt = createS3Event(validEmail);
+        var actualUrl = handler.handleRequest(s3EventNotFromSikt, CONTEXT);
+        assertThat(actualUrl, not(contains(UriWrapper.fromUri("https://sccontent-scudd-delivery-prod.s3.amazonaws.com/sccontent-scudd-delivery-prod/cristin_1/2023-6-14/cristin_1_2023-6-14_ANI-CITEDBY.zip?my-query-param=should-be-preserved").getUri())));
+    }
+
+    //TODO: implement download feature.
+//    @Test
+//    void shouldThrowExceptionIfTheURIsInTheEmailIsNotDownloadable() throws IOException {
+//        var s3EventNotFromSikt = createS3Event(validEmail);
+//        var exception = assertThrows(EmailException.class, () -> handler.handleRequest(s3EventNotFromSikt, CONTEXT));
+//        assertThat(exception.getMessage(), containsString(NO_URL_PRESENT_IN_MESSAGE));
+//        assertThat(exception.getObjectKey(), is(equalTo(extractObjectKey(s3EventNotFromSikt))));
+//        assertThat(exception.getBucket(), is(equalTo(INPUT_BUCKET_NAME)));
+//
+//    }
+
+    @Test
+    void shouldReturnASetOfDownloadsURIWhenTheEmailContainsTheURI() throws IOException {
+        var s3EventNotFromSikt = createS3Event(validEmail);
+        var actualUrl = handler.handleRequest(s3EventNotFromSikt, CONTEXT);
```

Comment:
notFromSikt? is it then direct from aws?

Reply:
whoops, copy paste code from above test and change which emailGenerator method it should use. Fixed now